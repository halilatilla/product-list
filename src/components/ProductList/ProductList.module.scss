.productList {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(252px, 1fr));
  gap: 14px;
}

.pagination {
  margin-top: 32px;
}
