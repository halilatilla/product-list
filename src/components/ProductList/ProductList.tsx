import { FC, useState } from 'react'
import classnames from 'classnames'

import { useAppSelector } from '@src/store'
import { Pagination } from '@src/components'
import { paginationOptions } from '@src/constants'
import { getListByPaginated, getSortedProductList } from '@src/lib'
import ProductCard from './ProductCard/ProductCard'
import styles from './ProductList.module.scss'

interface Props {
  className?: string
}

const ProductList: FC<Props> = ({ className, ...rest }) => {
  const { filteredProducts, sortingBy } = useAppSelector((state) => state.filter)
  const [page, setPage] = useState(paginationOptions.START_PAGE)

  const handlePagination = (e: number) => {
    setPage(e)
  }

  return (
    <div data-testid="productList">
      <div className={classnames(styles.productList, className)} {...rest}>
        {getListByPaginated(getSortedProductList(filteredProducts, sortingBy), page, paginationOptions.PAGE_SIZE)?.map(
          (product) => (
            <ProductCard key={product?.productId} product={product} />
          ),
        )}
      </div>
      <div className={styles.pagination}>
        <Pagination
          current={page}
          total={filteredProducts.length}
          pageSize={paginationOptions.PAGE_SIZE}
          onChange={handlePagination}
          hideOnSinglePage={true}
        />
      </div>
    </div>
  )
}

export default ProductList
