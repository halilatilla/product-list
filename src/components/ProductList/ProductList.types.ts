import { IProduct } from '@src/types'
export default interface IProductList {
  className?: string
  page: number
  products: IProduct[]
}
