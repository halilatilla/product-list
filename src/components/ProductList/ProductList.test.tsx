import { render, screen } from '@src/testUtils'

import ProductList from './ProductList'

describe('ProductList', () => {
  it('renders correctly', () => {
    const { container } = render(<ProductList />)
    expect(container).toMatchSnapshot()
  })

  it('renders ProductList', () => {
    render(<ProductList />)

    const productList = screen.getByTestId('productList')

    expect(productList).toBeInTheDocument()
  })
})
