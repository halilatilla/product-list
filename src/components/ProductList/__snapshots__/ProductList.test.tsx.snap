// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<div>
  <div
    data-testid="productList"
  >
    <div
      class="productList"
    />
    <div
      class="pagination"
    />
  </div>
</div>
`;
